lib/modules/bulk_data/bulk_data_authorization_sequence.rb

```diff
@@ -0,0 +1,277 @@
+# frozen_string_literal: true
+
+module Inferno
+  module Sequence
+    class BulkDataAuthorizationSequence < SequenceBase
+      title 'Bulk Data Authentication'
+
+      test_id_prefix 'BDA'
+
+      requires :bulk_client_id, :bulk_private_key, :bulk_token_endpoint
+
+      description 'Test Bulk Data Authorization Token Endpoint'
+
+      def authorize(bulk_private_key: @instance.bulk_private_key,
+                    content_type: 'application/x-www-form-urlencoded',
+                    scope: 'system/*.read',
+                    grant_type: 'client_credentials',
+                    client_assertion_type: 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
+                    iss: @instance.bulk_client_id,
+                    sub: @instance.bulk_client_id,
+                    aud: @instance.bulk_token_endpoint,
+                    exp: 5.minutes.from_now,
+                    jti: SecureRandom.hex(32))
+        jwt_token = JSON::JWT.new(
+          iss: iss,
+          sub: sub,
+          aud: aud,
+          exp: exp,
+          jti: jti
+        ).compact
+
+        jwk = JSON::JWK.new(JSON.parse(bulk_private_key))
+
+        jwt_token.header[:kid] = jwk['kid']
+        jwk_private_key = jwk.to_key
+        client_assertion = jwt_token.sign(jwk_private_key, 'RS384')
+
+        payload =
+          {
+            'scope' => scope,
+            'grant_type' => grant_type,
+            'client_assertion_type' => client_assertion_type,
+            'client_assertion' => client_assertion.to_s
+          }.compact
+
+        header =
+          {
+            content_type: content_type,
+            accept: 'application/json'
+          }.compact
+
+        uri = Addressable::URI.new
+        uri.query_values = payload
+
+        LoggedRestClient.post(@instance.bulk_token_endpoint, uri.query, header)
+      end
+
+      test :require_content_type do
+        metadata do
+          id '01'
+          name 'Bulk Data authorization request shall use content_type "application/x-www-form-urlencoded"'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/authorization/index.html#protocol-details'
+          description %(
+            After generating an authentication JWT, the client requests a new access token via HTTP POST to the FHIR authorization server’s token endpoint URL, using content-type application/x-www-form-urlencoded
+          )
+        end
+
+        response = authorize(content_type: 'application/json')
+        assert_response_bad(response)
+      end
+
+      test :require_system_scope do
+        metadata do
+          id '02'
+          name 'Bulk Data authorization request shall use "system" scope'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/authorization/index.html#scopes'
+          description %(
+            clients SHALL use “system” scopes.
+
+            System scopes have the format system/(:resourceType&#124;&#42;).(read&#124;write&#124;&#42;)
```

Comment:
```suggestion
            System scopes have the format `system/(:resourceType&#124;&#42;).(read&#124;write&#124;&#42;)`
```

We should format the scope format.